service/src/test/java/com/kodemon/service/test/BadgeServiceTest.java

```diff
@@ -0,0 +1,177 @@
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import javax.inject.Inject;
+
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.mockito.ArgumentMatchers.isNotNull;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Test class for Badge service
+ *
+ * @author Matej Poklemba
+ */
+@ContextConfiguration(classes = ServiceConfig.class)
+public class BadgeServiceTest extends AbstractTestNGSpringContextTests {
+    @Mock
+    private BadgeDao badgeDao;
+
+    @Inject
+    @InjectMocks
+    private BadgeService badgeService;
+
+    private Trainer trainer;
+    private Gym gym;
+    private Badge badge;
+    private String badgename = "Best Badge";
+
+    @BeforeClass
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @BeforeMethod
+    public void prepare() {
+        trainer = new Trainer();
+        trainer.setUserName("brock1999");
+        trainer.setFirstName("Brock");
+        trainer.setLastName("Brockowski");
+        Date born = new Calendar.Builder().setDate(1999, 5, 5).build().getTime();
+        trainer.setDateOfBirth(born);
+
+        gym = new Gym(trainer);
+        gym.setCity("Saffron City");
+        gym.setType(PokemonType.WATER);
+        gym.setTrainer(trainer);
+
+        badge = new Badge(gym, trainer);
+        badge.setName(badgename);
+    }
+
+    @AfterMethod
+    public void reset_mocks() {
+        Mockito.reset(badgeDao);
+    }
+
+    @Test
+    public void createBadgeOfGymTest(){
+        Badge result = badgeService.createBadgeOfGym(gym);
+        assertThat("badge == null", result, is(notNullValue()));
+        assertThat(result.getGym(), is(gym));
+    }
+
+    @Test
+    public void assignTrainerToBadgeTest() {
+        when(badgeDao.save(badge)).thenReturn(badge);
+
+        Trainer trainer2 = new Trainer();
+        trainer2.setUserName("asdf");
+        trainer2.setFirstName("John");
+        trainer2.setLastName("Johnson");
+        Date born = new Calendar.Builder().setDate(1995, 1, 1).build().getTime();
+        trainer2.setDateOfBirth(born);
+
+        badgeService.assignTrainerToBadge(trainer2, badge);
+        assertThat(badge.getTrainer(), is(trainer2));
+        verify(badgeDao).save(badge);
+    }
+
+    @Test(expectedExceptions = {NullPointerException.class})
+    public void assignNullTrainerToBadgeTest() {
+        when(badgeDao.save(badge)).thenThrow(NullPointerException.class);
+        badgeService.assignTrainerToBadge(null, badge);
+        verify(badgeDao).save(badge);
```

Comment:
I think that `verify` here is not necessarry, as you are verifying that method was called by catching an exception, but leave it here.. id does nothing wrong.